tests/test_cli.py

```diff
@@ -468,37 +435,33 @@ def test_sync_command_finds_pyproject_in_parent_directory(clean_test_dir):
     )
 
     # Verify the vendor directory was created in the parent directory (where pyproject.toml is)
-    TEST_SRC_VENDOR = TEST_DIR / "python_modules"
+    TEST_SRC_VENDOR = test_dir / "python_modules"
     assert TEST_SRC_VENDOR.exists(), (
         f"python_modules directory was not created at {TEST_SRC_VENDOR}"
     )
 
     # Verify the .venv-workers directory was created in the parent directory
-    TEST_VENV_WORKERS = TEST_DIR / ".venv-workers"
+    TEST_VENV_WORKERS = test_dir / ".venv-workers"
     assert TEST_VENV_WORKERS.exists(), (
         f".venv-workers directory was not created at {TEST_VENV_WORKERS}"
     )
 
 
-def test_sync_recreates_venv_on_python_version_mismatch(clean_test_dir):
+def test_sync_recreates_venv_on_python_version_mismatch(test_dir):
     """
     Test that the sync command recreates the venv if the Python version
     mismatches, using real system commands.
     """
     # Create initial files in the clean test directory
-    create_test_pyproject([])
-
-    project_root = Path.cwd().resolve()
+    create_test_pyproject(test_dir)
 
-    sync_cmd = ["uvx", "--from", project_root, "pywrangler", "sync"]
-    venv_path = clean_test_dir / ".venv-workers"
+    sync_cmd = ["uv", "run", "pywrangler", "sync"]
```

Comment:
Hmmm, how can we be sure this isn't just installing `pywrangler` and running the installed version rather than the one we are testing?